scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/mirrors/OfflineMirror.scala

```diff
@@ -0,0 +1,56 @@
+package scala.meta
+package internal
+package semantic
+package mirrors
+
+import java.io._
+import org.scalameta.data._
+import scala.{Seq => _}
+import scala.collection.immutable.Seq
+import scala.collection.mutable
+import scala.meta.internal.io.PlatformIO
+import scala.meta.parsers._
+import scala.meta.semantic._
+
+@data class OfflineMirror(classpath: String, sourcepath: String) extends CommonMirror {
```

Comment:
Looks plausible. I don't think that we need `dialect` though, because it can be inferred from `sources`. If it can't (because sources are empty), then we can just pick whatever.